peps/pep-0752.rst

```diff
@@ -0,0 +1,399 @@
+PEP: 752
+Title: Package repository namespaces
+Author: Ofek Lev <ofekmeister@gmail.com>
+Sponsor: Barry Warsaw <barry@python.org>
+Status: Draft
+Type: Standards Track
+Topic: Packaging
+Created: 13-Aug-2024
+
+Abstract
+========
+
+This PEP specifies a way for organizations to reserve package name prefixes
+for future uploads.
+
+    "Namespaces are one honking great idea -- let's do more of
+    those!" - :pep:`20`
+
+Motivation
+==========
+
+The current ecosystem lacks a way for projects with many packages to signal a
+verified pattern of ownership. Some examples:
+
+* `Typeshed <https://github.com/python/typeshed>`__ is a community effort to
+  maintain type stubs for various packages. The stub packages they maintain
+  mirror the package name they target and are prefixed by ``types-``. For
+  example, the package ``requests`` has a stub that users would depend on
+  called ``types-requests``.
+* Major cloud providers like Amazon, Google and Microsoft have a common prefix
+  for each feature's corresponding package [1]_. For example, most of Google's
+  packages are prefixed by ``google-cloud-`` e.g. ``google-cloud-compute`` for
+  `using virtual machines <https://cloud.google.com/products/compute>`__.
+* Many projects support a model where some packages are officially maintained
+  and third-party developers are encouraged to participate by creating their
+  own.
+
+  - `MkDocs <https://github.com/mkdocs/mkdocs>`__ is a documentation framework
+    based on Markdown files. They have the concept of
+    `plugins <https://www.mkdocs.org/dev-guide/plugins/>`__ which may be
+    developed by anyone and by convention are prefixed by ``mkdocs-``.
+  - `Project Jupyter <https://jupyter.org>`__ is devoted to the development of
+    tooling for sharing interactive documents. They support `extensions`__
+    which in most cases (and in all cases for officially maintained extensions)
+    are prefixed by ``jupyter-``.
+  - `Datadog <https://www.datadoghq.com>`__ offers observability as a service
+    for organizations at any scale. The
+    `Datadog Agent <https://docs.datadoghq.com/agent/>`__ ships out-of-the-box
+    with
+    `official integrations <https://github.com/DataDog/integrations-core>`__
+    for many products, like various databases and web servers, which are
+    distributed as Python packages that are prefixed by ``datadog-``. There is
+    support for creating `third-party integrations`__ which customers may run.
+
+__ https://jupyterlab.readthedocs.io/en/stable/user/extensions.html
+__ https://docs.datadoghq.com/developers/integrations/agent_integration/
```

Comment:
This list feels too long for "*Motivation*", perhaps shorten or move somewhere else?

Reply:
I moved the nested list into a footnote.

No, only organizations may and I changed the first few words of the specification to make that explicit.